.circleci/config.yml

```diff
@@ -1,24 +1,40 @@
 version: 2
 
 jobs:
-
   build:
     docker:
-      - image: python:3.7-alpine
+      - image: circleci/python:3.7
     steps:
       - checkout
-      - run: apk add git
-      - run: pip install -r requirements.txt
-      - run: echo 'Run the tests....'
+
+      - run: sudo chown -R circleci:circleci /usr/local/bin
+      - run: sudo chown -R circleci:circleci /usr/local/lib/python3.7/site-packages
+      - restore_cache:
+          key: deps9-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
+
+      - run:
+          command: |
+            sudo pip install pipenv
+            pipenv install --dev
+
+      - save_cache:
+          key: deps9-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
+          paths:
+            - ".venv"
+            - "/usr/local/bin"
+            - "/usr/local/lib/python3.7/site-packages"
+
+      - run: pipenv run bash ./bin/lint.sh
+      - run: pipenv run bash ./bin/coverage.sh
+      - store_artifacts:
+          path: ./htmlcov
+          destination: coverage
```

Comment:
This saves the output from running the code coverage tool, which makes it easy to see what lines are are aren't covered: https://85-168994705-gh.circle-artifacts.com/0/coverage/index.html

It's under the `artifacts` button in on a build in Circle